### DISCUSSION TITLE
Material Variants - AR modes compatibility

### DISCUSSION THREAD

**Discussion by maciekglowka** [google/model-viewer#3316] [Q&A]
Hi, 

I am thinking about switching to material variants (instead of separate files, as I found out how to automate it in Blender).

My biggest doubt is compatibility though. Does it work fine in the scene-viewer and QuickLook (auto generated USDZ) ?
I am guessing a blob is being sent i both cases? So it shooould work? Any problems reported with that?

Maciek

**Comment by elalish** [comment]
You can test it on various devices here: https://modelviewer.dev/examples/scenegraph/#variants

Works great in WebXR (you can also swap variants during an AR session). Scene Viewer does not support this as it downloads the GLB URL separately, but any phone that supports Scene Viewer also supports WebXR. Quick Look with our USDZ auto-generation does work (no switching within AR, but it'll send the currently selected version to AR).

**Comment by maciekglowka** [comment]
Thanks @elalish !

That's a bummer that Scene Viewer does not support blobs. I think that will be a blocker for me :/
Devices do support WebXR but many browsers don't.  Webview holds strong..

![obraz](https://user-images.githubusercontent.com/25125159/160465763-8b7695cf-0166-477c-aa0d-037301feea17.png)

**Reply by elalish** [reply]
Yeah, true. WebXR *is* supported in WebLayer, but it will take some time for all the WebView users to switch over.

**Reply by maciekglowka** [reply]
With Android segmentation it will take ages. I think sooner Scene Viewer might get blobs :)

**Reply by elalish** [reply]
Sadly, I think not, as that's a security problem.

**Reply by allenwixted** [reply]
Any suggestions for using material variants while retaining animations for USDZ?

**Reply by elalish** [reply]
No, this is the problem with Apple not supporting either WebXR or glTF. There's not much we can do but ask nicely. (FYI @grorg).
